### DISCUSSION TITLE
Override behavior with multiple matches

### DISCUSSION THREAD

**Discussion by redbmk** [biomejs/biome#3166] [General]
I'm not sure if this is a bug so starting it out as a discussion.

I expected to be able to add multiple blocks of "overrides" and have items that have multiple matches pick up all of the overrides.

Here's a simplified example of my setup:

```jsonc
{
  "$schema": "https://biomejs.dev/schemas/1.7.1/schema.json",
  "linter": {
    "enabled": true,
    "rules": {
      "recommended": true,
      "correctness": { "all": true, "noUndeclaredVariables": "error" },
      "suspicious": { "noConsoleLog": "warn" }
    }
  },
  "overrides": [
    { "include": ["scripts/k6.js"], "javascript": { "globals": ["__ENV", "__VU"] } },
    {
      "include": ["scripts"],
      "linter": { "rules": { "suspicious": { "noConsoleLog": "off" } } }
    }
  ]
}
```

and lets say somewhere in `scripts/k6.js` I have `console.log({ __ENV, __VU })`

If I have it set up like this, then `scripts/k6.js` throws errors that `__ENV` and `__VU` are undeclared. If I swap the order of the two override sections, it picks up on the globals but then throws warnings that I'm using `console.log`.

I'm having trouble finding documentation on how exactly the `overrides` work.

Is there a way I can update the overrides so that `scripts/k6.js` has both those overrides without having to repeat the noConsoleLog override?

**Comment by Conaclos** [comment]
I think the issue is coming from the glob. Did you try `scripts/**` instead of `scripts` in the second override?

**Reply by redbmk** [reply]
Hmm, that doesn't seem to make a difference. It still just takes whichever one comes last for that file and ignores the overrides from previous matches.

**Reply by redbmk** [reply]
Not sure why the discussion doesn't mention the related issues, but in case anyone else stumbles across this from an older version, I just noticed that you filed https://github.com/biomejs/biome/issues/3176 from this and then promptly fixed it in https://github.com/biomejs/biome/pull/3183 🎉 

Thanks for the super quick turnaround!

Edit: Oh I see it now on the right hand side under "Events".
